### PR TITLE
feat: add mistral as a new guardrail provider

### LINKED ISSUE #755: [Feature] Add Mistral Moderations API as a guardrail Provider
### What Would You Like to See with the Gateway?

Add mistral moderations api as a guardrail provider.

### Context for your Request

_No response_

### Your Twitter/LinkedIn

_No response_

### PR DESCRIPTION
**Pull request by b4s36t4**
**Title:** 
- Mistral Moderations support for guardrail plugins.

**Related Issues:** (optional)
- /closes #755

### CODE REVIEW

**Comment by b4s36t4** [comment]
Example curl Request:

```sh
curl --request POST \
  --url http://localhost:8787/v1/chat/completions \
  --header 'authorization: <api_key>' \
  --header 'content-type: application/json' \
  --header 'x-portkey-config: {"before_request_hooks": [ {"type":"guardrail", "name": "test", "checks": [ {"id": "mistral.pii", "parameters": { "credentials": {"apiKey":"<mistral_api_key>" }}} ]} ]}' \
  --header 'x-portkey-provider: groq' \
  --data '{
  "messages": [
    {
      "role": "user",
      "content": "Say Hi. My name is Jhon Doe and my email is user@example.com"
    }
  ]
}'
```

Supported guardrails are

> 'sexual', 'hate_and_discrimination', 'violence_and_threats', 'dangerous_and_criminal_content', 'selfharm', 'health',  'financial', 'law', 'pii',

File: plugins/mistral/manifest.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
+  "id": "mistral",
+  "description": "Mistral Content Moderation classifier leverages the most relevant policy categories for effective guardrails and introduces a pragmatic approach to LLM safety by addressing model-generated harms such as unqualified advice and PII",
+  "credentials": {
```
Comment (narengogi) [inline plugins/mistral/manifest.json:4]: "model" is not mentioned in this schema, but is referenced in the plugin code

**Review by narengogi — DISMISSED** [review]

File: plugins/mistral/manifest.json (hunk 1 above)
Comment (b4s36t4) [reply plugins/mistral/manifest.json:4]: Mistral doesn't have multiple models for guardrailing, having it here might pass give wrong impression, just added condition for future reference (for now model is completely optional).

**Comment by VisargD** [comment]
Hey @b4s36t4 - Can you please add tests as well like other plugins

**Review by narengogi — DISMISSED** [review]

File: plugins/mistral/mistra.test.ts
Comment (VisargD) [inline plugins/mistral/mistra.test.ts:1]: typo in file name

File: src/middlewares/hooks/index.ts [hunk 2]
```diff
@@ -261,7 +261,8 @@ export class HooksManager {
         context,
         check.parameters,
         eventType,
-        options
+        options,
+        fn
```
Comment (VisargD) [inline src/middlewares/hooks/index.ts]: We can remove this arg because it is no longer required with the new approach.

File: plugins/index.ts [hunk 3]
```diff
@@ -80,4 +94,11 @@ export const plugins = {
     toxicity: patronustoxicity,
     custom: patronuscustom,
   },
+  mistral: mistralGuardCategories.reduce(
```
Comment (VisargD) [inline plugins/index.ts]: We can convert this to a single plugin like portkey.moderateContent and accept all the required categories in it

**Comment by VisargD** [comment]
Added 2 comments. Everything else LGTM

File: plugins/mistral/manifest.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                "hate_and_discrimination",
+                "violence_and_threats",
+                "dangerous_and_criminal_content",
+                "selfharm",
+                "health",
+                "financial",
+                "law",
+                "pii"
+              ],
+              "default": [
```
Comment (VisargD) [inline plugins/mistral/manifest.json:53]: Values in default do not match with the enum

File: plugins/mistral/manifest.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      }
+    },
+    "required": ["apiKey"]
+  },
+  "functions": [
+    {
+      "name": "Moderate Content",
+      "id": "moderateContent",
+      "type": "guardrail",
+      "supportedHooks": ["beforeRequestHook"],
```
Comment (VisargD) [inline plugins/mistral/manifest.json]: Is there a reason to only support beforeRequestHook?

Comment (b4s36t4) [reply plugins/mistral/manifest.json]: there's no specific reason, just replicated the same schema we have for the `portkey` plugin.

Comment (VisargD) [reply plugins/mistral/manifest.json]: That could be a bug for portkey plugin manifest. We can add it here because the handler is anyways supporting it.

**Review by VisargD — APPROVED** [review]

**Pull request merged**